lib/ansible/modules/cloud/vmware/vmware_guest_tools_wait.py

```diff
@@ -0,0 +1,204 @@
+from ansible.module_utils.vmware import connect_to_api, gather_vm_facts, vmware_argument_spec, find_vm_by_id
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+HAS_PYVMOMI = False
+try:
+    import pyVmomi
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    pass
+
+
+class PyVmomiHelper(object):
+    def __init__(self, module):
+        if not HAS_PYVMOMI:
+            module.fail_json(msg='pyvmomi module required')
+
+        self.module = module
+        self.params = module.params
+        self.content = connect_to_api(self.module)
+
+    def getvm(self, name=None, uuid=None, folder=None):
+        vm = None
+        match_first = False
+        if uuid:
+            vm = find_vm_by_id(self.content, vm_id=uuid, vm_id_type="uuid")
+        elif folder and name:
+            if self.params['name_match'] == 'first':
+                match_first = True
+            vm = find_vm_by_id(self.content, vm_id=name, vm_id_type="inventory_path", folder=folder, match_first=match_first)
+        return vm
+
+    def gather_facts(self, vm):
+        return gather_vm_facts(self.content, vm)
+
+    def wait_for_tools(self, vm, poll=100, sleep=5):
+        tools_running = False
+        vm_facts = {}
+        poll_num = 0
+        vm_uuid = vm.config.uuid
+        while not tools_running and poll_num <= poll:
+            newvm = self.getvm(uuid=vm_uuid)
+            vm_facts = self.gather_facts(newvm)
+            if vm_facts['guest_tools_status'] == 'guestToolsRunning':
+                tools_running = True
+            else:
+                time.sleep(sleep)
+                poll_num += 1
+
+        if not tools_running:
+            return {'failed': True, 'msg': 'VMware tools either not present or not running after {0} seconds'.format((poll * sleep))}
+
+        changed = False
+        if poll_num > 0:
+            changed = True
+        return {'changed': changed, 'failed': False, 'instance': vm_facts}
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        name_match=dict(type='str', default='first'),
+        folder=dict(type='str', default='/vm'),
+        uuid=dict(type='str'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_one_of=[['name', 'uuid']],
+        required_together=['name', 'folder']
+    )
+
+    # FindByInventoryPath() does not require an absolute path
+    # so we should leave the input folder path unmodified
+    module.params['folder'] = module.params['folder'].rstrip('/')
+
+    pyv = PyVmomiHelper(module)
+    # Check if the VM exists before continuing
+    vm = pyv.getvm(name=module.params['name'],
+                   folder=module.params['folder'],
+                   uuid=module.params['uuid'])
+
+    # VM exists
+    if vm:
+        try:
+            result = pyv.wait_for_tools(vm)
+            if result['failed']:
```

Comment:
Don't add stuff in a try-except block that should not fail.
If possible also catch specific exceptions with specific error messages.
(It is possible not all exceptions provide a workable string the same way, unfortunately)